src/log.cpp

```diff
@@ -62,21 +85,55 @@ void Log::log(LogLevel level, const char* msg, va_list args) {
     fflush(_file);
 }
 
+void Log::trace(const char* msg, ...) {
+    if (LOG_TRACE < _level) {
+        return;
+    }
+
+    va_list args;
+    va_start(args, msg);
+    log(LOG_TRACE, msg, args);
+    va_end(args);
+}
+
+void Log::debug(const char* msg, ...) {
+    if (LOG_DEBUG < _level) {
+        return;
+    }
+
+    va_list args;
+    va_start(args, msg);
+    log(LOG_DEBUG, msg, args);
+    va_end(args);
+}
+
 void Log::info(const char* msg, ...) {
+    if (LOG_INFO < _level) {
+        return;
+    }
+
     va_list args;
     va_start(args, msg);
     log(LOG_INFO, msg, args);
     va_end(args);
 }
 
 void Log::warn(const char* msg, ...) {
+    if (LOG_WARN < _level) {
```

Comment:
`log` function does not only print to a file/console, but also issues a corresponding JFR event, which is useful regardless of `loglevel`.

Reply:
I've moved that check into `Log::log`